cscs-checks/microbenchmarks/osu/osu_tests.py

```diff
@@ -72,6 +72,76 @@ def __init__(self):
         self.tags = {'diagnostic', 'ops'}
 
 
+@rfm.required_version('>=2.14')
+@rfm.parameterized_test(['production'])
+class AllreduceTest(rfm.RegressionTest):
+    def __init__(self, variant):
+        super().__init__()
+        self.strict_check = False
+        self.valid_systems = ['daint:gpu', 'dom:gpu']
+        self.descr = 'Allreduce OSU microbenchmark'
+        self.build_system = 'Make'
+        self.build_system.makefile = 'Makefile_allreduce'
+        self.executable = './osu_allreduce'
+        # The -x option controls the number of warm-up iterations
+        # The -i option controls the number of iterations
+        self.executable_opts = ['-m', '8', '-x', '1000', '-i', '20000']
+        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu',
+                                    'PrgEnv-intel']
```

Comment:
Why we need all these programming environments?

Reply:
reduced to PrgEnv-gnu only